crates/languages/src/typescript/highlights.scm

```diff
@@ -2,6 +2,23 @@
 
 (identifier) @variable
 
+; Special identifiers
+
+((identifier) @constructor
+ (#match? @constructor "^[A-Z]"))
+
+((identifier) @type
+ (#match? @type "^[A-Z]"))
```

Comment:
The second query always overwrites the first, can anyone explain why it's written in this way?

Reply:
This also causes CamelCase variables to be highlighted as types.